### PR TITLE
Completely remove need for denodeify

### PR DESCRIPTION
**Pull request by RabsRincon**
fs-extra already does what we were asking the package to do

I've opened the PR so @partouf can check it in case that this would cause some problems when merging #1503 (I have refrained from moving more stuff to async/await until that PR is closed so the merge goes smoother :) )

### CODE REVIEW

File: lib/cache/on-disk.js [hunk 1]
```diff
@@ -79,24 +75,20 @@ class OnDiskCache extends BaseCache {
             `totalling ${this.cache.length} bytes on disk`;
     }
 
-    getInternal(key) {
+    async getInternal(key) {
         const cached = this.cache.get(key);
         if (!cached) return Promise.resolve({hit: false});
```
Comment (partouf) [inline lib/cache/on-disk.js]: Can this not now also be return {hit: false}; ?

File: lib/base-compiler.js [hunk 2]
```diff
@@ -862,10 +859,9 @@ Please select another pass or change filters.`;
                 if (postProcess.length) {
                     return this.execPostProcess(result, postProcess, outputFilename, maxSize);
                 } else {
-                    return this.readFile(outputFilename).then(contents => {
-                        result.asm = contents.toString();
-                        return Promise.resolve(result);
-                    });
+                    const contents = await fs.readFile(outputFilename);
+                    result.asm = contents.toString();
+                    return Promise.resolve(result);
```
Comment (RabsRincon) [inline lib/base-compiler.js]: As with the earlier comment, could this be a `return result` now? Or am I missing something (Still on the process of fully grasping async/await)

Comment (apmorton) [reply lib/base-compiler.js]: It can be, yes.

Comment (RabsRincon) [reply lib/base-compiler.js]: Thanks! :)

**Pull request merged**